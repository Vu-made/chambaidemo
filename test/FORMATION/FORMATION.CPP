#include <bits/stdc++.h>
#define N 1000001
using namespace std;
int L[N], R[N];
int n, m;
void xuly1()
{
    int i,j;
    cin >> i >> j;
    L[i] = L[j];
    R[i] = j;
    R[L[j]] = i;
    L[j] = i;
}

void xuly2()
{
    int i,j;
    cin >> i >> j;
    L[i] = j;
    R[i] = R[j];
    L[R[j]] = i;
    R[j] = i;
}

void xuly3()
{
    int i;
    cin >> i;
    R[L[i]] = R[i];
    L[R[i]] = L[i];
}

void xuly4()
{
    int i;
    cin >> i;
    cout << L[i] << ' ' << R[i] << endl;
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    freopen("FORMATION.INP","r",stdin);
    freopen("FORMATION.OUT","w",stdout);
    cin >> n >> m;
    for (int i=1; i<=m; i++)
    {
        int k;
        cin >> k;
        if (k == 1) xuly1();
        else if (k==2) xuly2();
        else if (k==3) xuly3();
        else xuly4();
    }
    return 0;
}
