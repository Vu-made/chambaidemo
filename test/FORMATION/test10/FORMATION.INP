954 86
1 228 1
1 664 228
3 664
2 648 1
3 228
4 1
4 648
2 291 648
2 109 291
1 369 109
2 136 1
4 291
3 291
1 377 109
4 369
1 710 136
1 325 377
2 924 377
1 372 325
3 372
1 401 377
4 1
4 401
4 136
4 325
1 591 1
1 567 109
1 795 136
1 153 795
4 153
4 1
1 807 401
4 795
1 302 325
4 591
1 429 369
1 468 325
3 109
4 369
2 186 468
4 302
3 153
1 732 325
1 831 186
2 24 302
1 743 186
3 831
4 924
1 733 325
4 567
3 136
3 24
1 465 648
3 710
3 325
2 69 377
3 429
1 293 465
3 1
3 465
4 733
2 368 369
3 468
2 717 69
1 925 795
1 499 302
4 795
2 22 924
2 844 732
2 810 293
1 841 69
3 401
2 797 377
4 924
4 567
4 807
4 810
4 732
3 22
4 733
3 377
1 493 368
2 131 499
2 699 795
3 369
1 296 797
