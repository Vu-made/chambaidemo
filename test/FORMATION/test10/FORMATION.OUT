0 648
1 0
648 369
648 377
0 710
325 377
710 648
369 401
710 795
591 710
153 136
0 1
429 302
369 468
377 567
924 0
732 807
925 293
717 22
22 0
733 377
293 648
186 844
844 807
